apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/RdaApiClaimMessageMetaData.java

```diff
@@ -0,0 +1,84 @@
+package gov.cms.bfd.model.rda;
+
+import java.time.Instant;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.experimental.FieldNameConstants;
+
+@Entity
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
+@FieldNameConstants
+@Table(name = "`RdaApiClaimMessageMetaData`", schema = "`pre_adj`")
+public class RdaApiClaimMessageMetaData {
+  private static final String SequenceName = "`rda_api_claim_message_meta_data_meta_data_id_seq`";
+
+  /**
+   * We insert a record every time a change is received from RDA API so the primary key has to be
+   * unique even if we receive the same message twice.
+   */
+  @Id
+  @Column(name = "`metaDataId`", nullable = false, updatable = false)
+  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = SequenceName)
+  @SequenceGenerator(
+      name = SequenceName,
+      schema = "`pre_adj`",
+      sequenceName = SequenceName,
+      allocationSize = 25)
+  @EqualsAndHashCode.Include
+  private long metaDataId;
+
+  /** Either FISS or MCS. */
+  @Enumerated(EnumType.STRING)
+  @Column(name = "`claimType`", length = 4, nullable = false)
+  @EqualsAndHashCode.Include
+  private RdaApiProgress.ClaimType claimType;
+
+  /**
+   * sequenceNumber from the RDA API change message object. Indicates the version of the claim that
+   * was written to the database.
+   */
+  @Column(name = "`sequenceNumber`", nullable = false)
+  @EqualsAndHashCode.Include
+  private long sequenceNumber;
+
+  /** Either dcn (FISS) or idrClmHdIcn (MCS). */
+  @Column(name = "`claimId`", length = 25, nullable = false)
+  private String claimId;
+
+  @ManyToOne(fetch = FetchType.EAGER)
+  @JoinColumn(name = "`mbiId`")
+  private Mbi mbiRecord;
+
+  /** Either currStatus (FISS) or idrStatusCode (MCS). */
+  @Column(name = "`claimState`", length = 1, nullable = false)
```

Comment:
we should allow NULL here, particularly for MCS records

Reply:
I have made the column nullable.